llvm/lib/Target/AIE/AIEBaseAliasAnalysis.cpp

```diff
@@ -77,6 +83,36 @@ static bool isAIEPtrAddIntrinsic(Intrinsic::ID ID, unsigned &InPtrIdx) {
   return false;
 }
 
+/// This gives all indexes for read only operands ie.
+/// excluding counters.
+static SmallVector<unsigned, 5> getAddIntrinsicOps(Intrinsic::ID ID) {
+  switch (ID) {
+  case Intrinsic::aie2_add_2d:
+    return {1, 2, 3 /*,count*/};
+  case Intrinsic::aie2_add_3d:
+    return {1, 2, 3, 4, /*,count1*/ 6,
+            /*,count2*/};
+  default:
+    llvm_unreachable("Unknown intrinsic");
+  }
+
+  return {};
+}
+
+/// This gives all indexes counter operands.
+static SmallVector<unsigned, 5> getAddIntrinsicCounterOps(Intrinsic::ID ID) {
```

Comment:
nit: it looks as if `getAddIntriscOps` is complementary to `getAddIntrinsicCounterOps`, so can be computed as `allInputOperands - getAddIntrinsicCounterOps`